### PR TITLE
Expose interface for the client to control identity

### PR DESCRIPTION
**Pull request by sourishkrout**
1. Minor rename; it seems like go packages favor entity-/data-centric names over operations-centric ones.
2. Introducing a way to turn doc/cell identity-tracking off selectively. I will handle the client side (extension).

Could you please fill in the gap on the serializer side, @pastuxso? Hope this won't take up too much time (cc @joseeantonior).

### CODE REVIEW

File: internal/api/runme/parser/v1/parser.proto [hunk 1]
```diff
@@ -51,6 +58,7 @@ message DeserializeResponse {
 
 message SerializeRequest {
   Notebook notebook = 1;
+  RunmeIdentity identity = 2;
```
Comment (sourishkrout) [inline internal/api/runme/parser/v1/parser.proto:61]: @pastuxso I'm pondering if we should have an "options" field here that "identity" will be part of. Make the distinction between message/data and config/options more obvious; what do you think?

File: internal/document/editor/editorservice/service.go [hunk 2]
```diff
@@ -70,8 +71,13 @@ func (s *parserServiceServer) Serialize(_ context.Context, req *parserv1.Seriali
 
 	cells := make([]*editor.Cell, 0, len(req.Notebook.Cells))
 	for _, cell := range req.Notebook.Cells {
-		if _, ok := cell.Metadata["id"]; !ok && cell.Kind == parserv1.CellKind_CELL_KIND_CODE {
-			cell.Metadata["id"] = idgen.GenerateID()
+		if req.Identity.Number() == parserv1.RunmeIdentity_RUNME_IDENTITY_ALL.Number() ||
```
Comment (sourishkrout) [inline internal/document/editor/editorservice/service.go:74]: I'm fairly certain you can compare the props without calling `Number()` on it, no?

**Review by sourishkrout — COMMENTED** [review]
See remark on enums.

File: internal/api/runme/parser/v1/parser.proto (hunk 1 above)
Comment (pastuxso) [reply internal/api/runme/parser/v1/parser.proto:61]: I like the idea. Are you thinking of something like a **`Params`** structure?"

Comment (sourishkrout) [reply internal/api/runme/parser/v1/parser.proto:61]: > @sourishkrout I like the idea, Are you thinking of something like a **`Params`** structure?"

As discussed out-of-band, let's go with `Options`.

Comment (pastuxso) [reply internal/api/runme/parser/v1/parser.proto]: Done :white_check_mark:

**Review by pastuxso — APPROVED** [review]

**Pull request merged**